app/io/pathfinder/controllers/ClusterController.java

```diff
@@ -0,0 +1,104 @@
+package io.pathfinder.controllers;
+
+import com.avaje.ebean.Ebean;
+import com.avaje.ebean.text.json.JsonContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import javassist.compiler.SymbolTable;
+import play.libs.Json;
+import play.mvc.Controller;
+import io.pathfinder.models.Cluster;
+import play.mvc.Result;
+import java.util.Iterator;
+
+import javax.persistence.PersistenceException;
+
+public class ClusterController extends Controller {
+    private JsonContext jsonContext = Ebean.createJsonContext();
+
+    public Result getClusters() {
+        return ok(jsonContext.toJson(Cluster.find.all()));
+    }
+
+    public Result getCluster(long id) {
+        Cluster cluster = Cluster.find.byId(id);
+
+        if (cluster == null) {
+            return notFound();
+        }
+
+        return ok(jsonContext.toJson(cluster));
+    }
+
+    public Result createCluster() {
+        JsonNode json = request().body().asJson();
+        if (json == null) {
+            return badRequest("Expected content-type text/json or application/json");
+        }
+
+        Cluster cluster;
+
+        try {
+            cluster = Json.fromJson(json, Cluster.class);
+
+            cluster.save();
+            return created(jsonContext.toJson(cluster));
+        } catch (PersistenceException e) {
+            return internalServerError("Error saving cluster to the database: " + e.getMessage());
+        } catch (Exception e) {
+            return badRequest("Unable to map json to commodity object: " + e.getMessage());
+        }
+    }
+
+    public Result editCluster(long id) {
+        Cluster cluster = Cluster.find.byId(id);
+
+        if (cluster == null) {
+            return notFound();
+        }
+
+        ObjectNode clusterJson = (ObjectNode) Json.toJson(cluster);
+        ObjectNode body;
+
+        try {
+            JsonNode json = request().body().asJson();
+            body = (ObjectNode) json;
+        } catch (ClassCastException e) {
+            return badRequest("Cannot cast request body to ObjectNode: " + e.getMessage());
```

Comment:
I'm not referring to the tests. This is the Cluster Controller. I'm referring to the fact that the mobile SDK will hit this endpoint, and may potential get this response, which would not be helpful from the perspective of the mobile SDK.